### PR TITLE
Adding and deleting bikes

### LINKED ISSUE #5: Add a form to add and remove bikes

### PR DESCRIPTION
**Pull request by jankupis**

### CODE REVIEW

File: src/components/AddBikeDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  })
+);
+
+export interface AddBikeDialogProps {
+  setBikes: (value: React.SetStateAction<Bike[]>) => void;  
+}
+
+const AddBikeDialog = (props: AddBikeDialogProps) => {
+    const classes = useStyles()
+  const [open, setOpen] = React.useState(false);
```
Comment (chlebnymaciej) [inline src/components/AddBikeDialog.tsx]: please add type

File: src/components/AddBikeDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+          <FormControl className={classes.formControl}>
+        <InputLabel id="demo-simple-select-label">Station</InputLabel>
+        <Select
+          labelId="demo-simple-select-label"
+          id="demo-simple-select"
+          value={stationId}
+          onChange={handleChange}
+        >
+          {
+            stations.map((station, _) =>
```
Comment (chlebnymaciej) [inline src/components/AddBikeDialog.tsx]: just (station) will work

**Review by chlebnymaciej — COMMENTED** [review]
please apply those changes

File: src/components/AddBikeDialog.tsx (hunk 1 above)
Comment (jankupis) [reply src/components/AddBikeDialog.tsx]: Done

File: src/components/AddBikeDialog.tsx (hunk 2 above)
Comment (jankupis) [reply src/components/AddBikeDialog.tsx]: Done

File: src/components/AddBikeDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    setOpen(false);
+  };
+
+  useEffect(() => {
+    getAllStations().then(res => {
+      if (res.isError) {
+          enqueueSnackbar("Could not retrive stations", { variant: "error" });
+          return;
+      }
+      setStations((res.data || []).map(x => x));
```
Comment (ArturChmura) [inline src/components/AddBikeDialog.tsx]: what is 'map' for?

File: src/components/AddBikeDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        <DialogTitle id="form-dialog-title">Create bike</DialogTitle>
+        <DialogContent>
+          <DialogContentText>
+            Select a station to assign the bike to.
+          </DialogContentText>
+          <FormControl className={classes.formControl}>
+        <InputLabel id="demo-simple-select-label">Station</InputLabel>
+        <Select
+          labelId="demo-simple-select-label"
+          id="demo-simple-select"
```
Comment (ArturChmura) [inline src/components/AddBikeDialog.tsx]: change to proper values

File: src/components/AddBikeDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    bike.then(response => {
+      if(response.isError)
+      {
+        enqueueSnackbar("Could not add bike", { variant: "error" });
+      }
+      else if(response.data)
+      {
+        props.setBikes(prev => 
+          response.data ? 
+          prev = [...prev, response.data] : prev
```
Comment (ArturChmura) [inline src/components/AddBikeDialog.tsx]: response.data will always return true, because you are checking it in elseif above

Comment (ArturChmura) [reply src/components/AddBikeDialog.tsx]: `prev = [...prev, response.data]` is wrong. Make it `[...prev, response.data]`.

**Review by ArturChmura — CHANGES_REQUESTED** [review]

File: src/components/AddBikeDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        )
```
Comment (ArturChmura) [inline src/components/AddBikeDialog.tsx]:
```suggestion
        const data = response.data;
        props.setBikes(prev => [...prev, data])
```

**Review by ArturChmura — CHANGES_REQUESTED** [review]

File: src/components/AddBikeDialog.tsx (hunk 5 above)
Comment (ArturChmura) [reply src/components/AddBikeDialog.tsx]: look at my suggestion below

File: src/components/AddBikeDialog.tsx (hunk 3 above)
Comment (ArturChmura) [reply src/components/AddBikeDialog.tsx]:
```suggestion
      setStations(res.data || []);
```

**Comment by jankupis** [comment]
Applied all suggestions, thanks

**Review by ArturChmura — APPROVED** [review]
Nicely done

**Pull request merged**